### PR TITLE
Prevent infinite retries

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

File: src/main/scala/cromwell/engine/backend/jes/JesBackend.scala [hunk 1]
```diff
@@ -110,6 +111,11 @@ object JesBackend {
     }
   }
 
+  def isFatalJESException(t: Throwable): Boolean = t match {
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: JesException

File: src/main/scala/cromwell/util/TryUtil.scala [hunk 2]
```diff
@@ -15,6 +16,7 @@ case class AggregatedException(exceptions: Seq[Throwable], prefixError: String =
 }
 
 object TryUtil {
+
```
Comment (geoffjentry) [inline src/main/scala/cromwell/util/TryUtil.scala]: Please no :)

Comment (Horneth) [inline src/main/scala/cromwell/util/TryUtil.scala]: Oops

Comment (cjllanwarne) [inline src/main/scala/cromwell/util/TryUtil.scala]: https://www.youtube.com/watch?v=WWaLxFIVX1s

Comment (geoffjentry) [inline src/main/scala/cromwell/util/TryUtil.scala]: https://www.youtube.com/watch?v=aG8ylfUd2_A

**Comment by mcovarr** [comment]
also @kshakir

File: src/main/scala/cromwell/engine/package.scala [hunk 3]
```diff
@@ -43,6 +43,8 @@ package object engine {
   type CallOutputs = Map[LocallyQualifiedName, CallOutput]
   type HostInputs = Map[String, WdlValue]
 
+  class CromwellFatalException(exception: Throwable) extends Throwable(exception)
```
Comment (kshakir) [inline src/main/scala/cromwell/engine/package.scala]: This should extend `Exception`, or even `RuntimeException`, but not `Throwable`.

**Comment by kshakir** [comment]
_Side note: Hoping our PO prioritizes testing soon. We're about to cross the 70% threshold._

Minor updates and then :+1:

**Comment by scottfrazer** [comment]
:+1:

**Comment by mcovarr** [comment]
Another vote for prioritized testing, though the increase in test coverage wouldn't automatically be reflected here.

**Comment by Horneth** [comment]
Added a test, it doesn't really test the full bug fix but at least that we throw `CromwellFatalException`s when a retry exhausts all its tries.

**Pull request merged**